CovidShield/mobile
# Fix submission flow on Android

## What
This PR fixes submission flow on Android and polishes the bridge.

Resolves https://github.com/CovidShield/mobile/issues/123

## How to achieve
Cherry-pick js changes in here https://github.com/CovidShield/mobile/pull/102 and resolve broken schema. 

## How to test
- Checkout `https://github.com/CovidShield/server`, make sure all dependencies are installed (check dev.ym#up), and run the server.
- Use `ngrok` to tunnel server endpoint
```
ngrok http 8000
ngrok http 8001
```
- In this PR, changes `.env` to (use your ngrok urls)
```
APP_ID_IOS=app.covidshield
APP_ID_ANDROID=com.google.android.apps.exposurenotification
APP_VERSION_CODE=1
APP_VERSION_NAME=1.0

SUBMIT_URL=https://2c4c45ae1a78.ngrok.io
RETRIEVE_URL=https://2a78ac7e6310.ngrok.io
HMAC_KEY=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
REGION=302

TEST_MODE=true
MOCK_SERVER=false
```
- Retrieve submission key with `curl -X POST -H "Authorization: Bearer test" https://2c4c45ae1a78.ngrok.io/new-key-claim`
- Run this PR on Android and test submission flow.

## Demo

https://drive.google.com/file/d/1R7YRvMqK3Ufe0rx_klK-9s5Ai8K5hV9y/view?usp=sharing

## Outstanding
I cannot test on two devices because key retrieval doesn't work. It is blocked by backend PR https://github.com/CovidShield/server/pull/67
